evalml/tests/data_checks_tests/test_datetime_format_data_check.py

```diff
@@ -244,7 +244,7 @@ def test_datetime_format_data_check_multiple_errors():
         + ["2021-01-31", "2021-02-02", "2021-02-04"]
         + pd.date_range("2021-02-05", periods=90).tolist()
     )
-    X = pd.DataFrame({"dates": dates})
+    X = pd.DataFrame({"dates": dates}, dtype="datetime64[ns]")
```

Comment:
Why are all these necessary?

Reply:
The previous behavior left it as object types, which in pandas 2 raises an error in [this line](https://github.com/alteryx/woodwork/blob/732cf90f051cef0d65fa3a3ecfefecf9d07743b2/woodwork/statistics_utils/frequency_inference/_determine_duplicate_values.py#L25) of woodwork. By casting it as an explicit datetime object, we're able to still utilize this function.